src/Bridge/Doctrine/EventListener/PublishMercureUpdatesListener.php

```diff
@@ -0,0 +1,175 @@
+    {
+        if (null === $messageBus && null === $publisher) {
+            throw new InvalidArgumentException('A message bus or a publisher must be provided.');
+        }
+
+        $this->resourceClassResolver = $resourceClassResolver;
+        $this->iriConverter = $iriConverter;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->serializer = $serializer;
+        $this->messageBus = $messageBus;
+        $this->publisher = $publisher;
+        $this->expressionLanguage = $expressionLanguage ?? class_exists(ExpressionLanguage::class) ? new ExpressionLanguage() : null;
+        $this->reset();
+    }
+
+    /**
+     * Collects created, updated and deleted entities.
+     */
+    public function onFlush(OnFlushEventArgs $eventArgs)
+    {
+        $uow = $eventArgs->getEntityManager()->getUnitOfWork();
+
+        foreach ($uow->getScheduledEntityInsertions() as $entity) {
+            $this->storeEntityToPublish($entity, 'createdEntities');
+        }
+
+        foreach ($uow->getScheduledEntityUpdates() as $entity) {
+            $this->storeEntityToPublish($entity, 'updatedEntities');
+        }
+
+        foreach ($uow->getScheduledEntityDeletions() as $entity) {
+            $this->storeEntityToPublish($entity, 'deletedEntities');
+        }
+    }
+
+    /**
+     * Purges tags collected during this request, and clears the tag list.
+     */
+    public function postFlush()
+    {
+        try {
+            foreach ($this->createdEntities as $entity) {
+                $this->publishUpdate($entity, $this->createdEntities[$entity]);
+            }
+
+            foreach ($this->updatedEntities as $entity) {
+                $this->publishUpdate($entity, $this->updatedEntities[$entity]);
+            }
+
+            foreach ($this->deletedEntities as $iri => $targets) {
+                $this->publishUpdate($iri, $targets);
+            }
+        } finally {
+            $this->reset();
+        }
+    }
+
+    private function reset(): void
+    {
+        $this->createdEntities = new \SplObjectStorage();
+        $this->updatedEntities = new \SplObjectStorage();
+        $this->deletedEntities = [];
+    }
+
+    /**
+     * @param object $entity
+     */
+    private function storeEntityToPublish($entity, string $property): void
+    {
+        $resourceClass = $this->getObjectClass($entity);
+        if (!$this->resourceClassResolver->isResourceClass($resourceClass)) {
+            return;
+        }
+
+        $value = $this->resourceMetadataFactory->create($resourceClass)->getAttribute('mercure', false);
+        if (false === $value) {
+            return;
+        }
+
+        if (\is_string($value)) {
+            if (null === $this->expressionLanguage) {
+                throw new RuntimeException('The Expression Language component is not installed. Try running "composer require symfony/expression-language".');
```

Comment:
is this the proper place to throw this, or can it be done in the metadata factory somehow?